slowapi/extension.py

```diff
@@ -378,16 +378,18 @@ def _inject_headers(
 
                 if existing_retry_after_header is not None:
                     # might be in http-date format
-                    retry_after = parsedate_to_datetime(existing_retry_after_header)
+                    try:
+                        retry_after = parsedate_to_datetime(existing_retry_after_header)
```

Comment:
This header is supposed to follow RFC2616 (https://tools.ietf.org/html/rfc2616#section-14.37) so it actually has 2 possible formats: int or http-date. The initial code was clearly not correct, but I feel like your fix is not completely right either. I think we should be checking the format (if not None) and respond accordingly. What do you think?

Reply:
@laurentS To help close this out, I offered up a different solution [here](https://github.com/laurentS/slowapi/pull/60/files#diff-f4d6ec107cc36585f9f51cefae0c2626c336e7cc1647e8592503ca6f94a066c9R453-R472).  By shifting this logic into it's own method, I think we're capturing the intent of your comment here.